sdccc/src/main/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditions.java

```diff
@@ -431,6 +450,50 @@ static boolean verifyStatePresentAndAssociated(
 
             return valid;
         }
+
+        static final class EpisodicContextReportStateHandleObserver implements MdibAccessObserver {
+
+            private final String expectedHandle;
+            private final Object signal = new Object();
+
+            private EpisodicContextReportStateHandleObserver(final String expectedStateHandle) {
+                this.expectedHandle = expectedStateHandle;
+            }
+
+            @Subscribe
+            public void onUpdate(final ContextStateModificationMessage report) {
+                for (List<AbstractContextState> states : report.getStates().values()) {
+                    for (AbstractContextState state : states) {
+                        if (this.expectedHandle.equals(state.getHandle())) {
+                            synchronized (signal) {
```

Comment:
this will compete with awakened threads i.e. it can block already awakened threads

and since this might be called quite often in a short amount of time, it might delay the return from waitForStateUpdate

Reply:
I don't understand your comment. At this point, the report was received and was identified as being the one that we are waiting for. It would be highly unusual if this would happen more than once, although it is of course possible.

Also, at this point the code has to notify the waiting Thread. Hence it is neccessary to call signal.notifyAll() and this is only possible when synchronizing on the signal first. I hence do not see any other way to realize this functionality.

Reply:
Done.